src/test/java/uk/gov/hmcts/ccd/domain/service/common/AccessControlServiceFilterTest.java

```diff
@@ -1152,6 +1154,162 @@ void filterComplexCaseFieldChildrenByUpdateAccessWhenAllAccessIsOnCreate() {
+                .withField(caseViewField1)
+                .withWizardPage(newWizardPage()
+                        .withId("Page One")
+                        .withField(caseViewField1)
+                        .build()
+                               )
+                .build();
+
+            CaseEventTrigger eventTrigger = accessControlService.updateCollectionDisplayContextParameterByAccess(
+                caseEventTrigger,
+                USER_ROLES);
+
+            CaseViewField people = eventTrigger.getCaseFields().stream()
+                .filter(e -> e.getId().equals("People")).findFirst().get();
+
+            assertAll(
+                () -> assertNotNull(people),
+                () -> assertTrue(people.getDisplayContextParameter().contains("#COLLECTION(")),
+                () -> assertTrue(people.getDisplayContextParameter().contains("allowInsert")),
+                () -> assertTrue(people.getDisplayContextParameter().contains("allowDelete"))
+                     );
+        }
+
+        @Test
+        @DisplayName("Should set #COLLECTION() in DisplayContextParameter of a collection caseField when no ACL set")
+        void updateCollectionDisplayContextParameterWhenFieldHasNoCreateDeleteUpdateRoles() {
+
+            final CaseViewField caseViewField1 = aViewField()
+                .withId("People")
+                .withFieldType(aFieldType()
+                    .withId("G339483948")
+                    .withType(COLLECTION)
+                    .build())
+                .withACL(anAcl()
+                    .withRole(ROLE_IN_USER_ROLES)
+                    .withCreate(false)
+                    .withDelete(false)
+                    .withUpdate(false)
+                    .build())
+                .build();
+            caseViewField1.getFieldType().setCollectionFieldType(getPersonFieldType());
+
+            CaseEventTrigger caseEventTrigger = newCaseEventTrigger()
+                .withField(caseViewField1)
+                .withWizardPage(newWizardPage()
+                        .withId("Page One")
+                        .withField(caseViewField1)
+                        .build()
+                               )
+                .build();
+
+            CaseEventTrigger eventTrigger = accessControlService.updateCollectionDisplayContextParameterByAccess(
+                caseEventTrigger,
+                USER_ROLES);
+
+            CaseViewField people = eventTrigger.getCaseFields().stream()
+                .filter(e -> e.getId().equals("People")).findFirst().get();
+
+            assertAll(
+                () -> assertNotNull(people),
+                () -> assertTrue(people.getDisplayContextParameter().contains("#COLLECTION(")),
+                () -> assertFalse(people.getDisplayContextParameter().contains("allowInsert")),
+                () -> assertFalse(people.getDisplayContextParameter().contains("allowDelete"))
+                     );
+        }
+    }
+
     @Nested
```

Comment:
We should add few integration tests as well, especially one testing collection of complex of collection...

Reply:
Yes, can add few.